src/Framework/MSBuildEventSource.cs

```diff
@@ -0,0 +1,338 @@
+        /// Call this method to notify listeners of information of how a project file built.
+        /// </summary>
+        public void BuildProjectStart()
+        {
+            if (IsEnabled()) WriteEvent(5);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of the end of the specified event.
+        /// </summary>
+        public void BuildProjectStop()
+        {
+            if (IsEnabled()) WriteEvent(6);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of information relevant to resolving references.
+        /// </summary>
+        public void ComputeClosureStart()
+        {
+            if (IsEnabled()) WriteEvent(7);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of the end of the specified event.
+        /// </summary>
+        public void ComputeClosureStop()
+        {
+            if (IsEnabled()) WriteEvent(8);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of information relevant to resolving conditionals.
+        /// </summary>
+        /// <param name="condition">Relevant information about where in the run of the progam it is.</param>
+        public void EvaluateConditionStart(string condition)
+        {
+            if (IsEnabled()) WriteEvent(9, condition);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of the end of the specified event.
+        /// </summary>
+        /// <param name="condition">Relevant information about where in the run of the progam it is.</param>
+        public void EvaluateConditionStop(string condition)
+        {
+            if (IsEnabled()) WriteEvent(10, condition);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of how the project data was evaluated.
+        /// </summary>
+        /// <param name="file">Relevant information about where in the run of the progam it is.</param>
+        public void EvaluateStart(string file)
+        {
+            if (IsEnabled())
+            {
+                string projectFile = String.IsNullOrEmpty(file) ? "(null)" : file;
+                WriteEvent(11, String.Format(CultureInfo.CurrentCulture, "Evaluate Project {0} - Start", projectFile));
+            }
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of progress made on Evaluate.
+        /// </summary>
+        /// <param name="file">Relevant information about where in the run of the progam it is.</param>
+        /// <param name="phaseNum">The stage in Evaluate that the program has reached.</param>
+        public void EvaluatePhase(string file, long phaseNum)
+        {
+            if (IsEnabled())
+            {
+                string projectFile = String.IsNullOrEmpty(file) ? "(null)" : file;
+                WriteEvent(12, String.Format(CultureInfo.CurrentCulture, "Evaluate Project {0}", projectFile), phaseNum);
```

Comment:
You should avoid this sort of string formatting in EventSources.  It's expensive at collection time and isn't necessary.  The standard idiom here is to just write an event that takes a `string file` and  `long phaseNum` and just emit those directly into the trace.  What's happening here is understood because in addition to the arguments, you also get the event name which has the semantic meaning that you want to impart to the viewer of the trace.